go/vt/vttablet/tabletmanager/vreplication/vplayer.go

```diff
@@ -152,8 +152,21 @@ func (vp *vplayer) play(ctx context.Context) error {
 // The foreign_key_checks value for a transaction is determined by the 2nd bit (least significant) of the flags:
 // - If set (1), foreign key checks are disabled.
 // - If unset (0), foreign key checks are enabled.
-// updateFKCheck also updates the state for the first row event that this vplayer and hence the connection sees.
+// updateFKCheck also updates the state for the first row event that this vplayer, and hence the db connection, sees.
 func (vp *vplayer) updateFKCheck(ctx context.Context, flags2 uint32) error {
+	mustUpdate := false
+	if vp.vr.WorkflowSubType == int32(binlogdatapb.VReplicationWorkflowSubType_AtomicCopy) {
+		// If this is an atomic copy, we must update the foreign_key_checks state even when the vplayer runs during
+		// the copy phase, i.e., for catchup and fastforward.
+		mustUpdate = true
```

Comment:
In `atomic-copy` workflows we run with `foreign_key_checks=on` by default, but not for regular workflows. 

During catchup/fastforward, we may get a transaction that sets `foreign_key_checks=off`. So we need to set the state, otherwise the transactions will fail. If not running in `atomic_copy` the checks are off always.